packages/@aws-cdk/aws-lambda/lib/function-url.ts

```diff
@@ -0,0 +1,195 @@
+import { IResource, Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { IFunction } from './function-base';
+import { CfnUrl } from './lambda.generated';
+
+
+/**
+ * The auth types for a function url
+ */
+export enum FunctionUrlAuthType {
+  /**
+   * Restrict access to authenticated IAM users only
+   */
+  AWS_IAM = 'AWS_IAM',
+
+  /**
+   * Bypass IAM authentication to create a public endpoint
+   */
+  NONE = 'NONE',
```

Comment:
Does this require a `Permission` on the Function to make the Function anonymously invokable? If so, that either needs documentation OR we do it automatically (and I think I'd prefer the latter).